libraries/AP_NavEKF2/AP_NavEKF2_Control.cpp

```diff
@@ -120,7 +121,7 @@ void NavEKF2_core::setWindMagStateLearningMode()
             P[21][21] = bodyMagFieldVar.z;
         } else {
             // set the variances equal to the observation variances
-            for (uint8_t index=18; index<=21; index++) {
+            for (uint8_t index=16; index<=21; index++) {
```

Comment:
where did the start index shift from 18->16 come from?

Reply:
it is a bug fix. See the definition of struct state_elements